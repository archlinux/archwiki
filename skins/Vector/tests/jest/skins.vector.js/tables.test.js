const tables = require( '../../../resources/skins.vector.js/tables.js' ).init;

describe( 'tables', () => {
	test( 'wraps table with div', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<table class="wikitable">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
			</div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'wraps multiple table with div', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<table class="wikitable">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
				<table class="wikitable">
				<table class="wikitable">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
				<table class="wikitable">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table>
			</div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'doesnt wrap nested tables', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<table class="wikitable">
					<tbody>
						<tr><th>table table table</th></tr>
						<tr><td><table class="wikitable"><tbody><tr><th>table table table</th></tr></tbody></table><td></tr>
					</tbody>
				</table>
			</div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'doesnt wrap tables that are not wikitables', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<table>
					<tbody>
						<tr><th>table table table</th></tr>
						<tr><td><table><tbody><tr><th>table table table</th></tr></tbody></table><td></tr>
					</tbody>
				</table>
			<div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'doesnt wrap tables that already have noresize', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<div class="noresize">
					<table class="wikitable">
						<tbody>
							<tr><th>table table table</th></tr>
						</tbody>
					</table>
				</div>
			</div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'doesnt wrap tables that are already wrapped', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<div>
					<table class="wikitable">
						<tbody>
							<tr><th>table table table</th></tr>
						</tbody>
					</table>
				</div>
			</div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'doesnt wrap floated tables', () => {
		document.body.innerHTML = `
			<div class="mw-parser-output">
				<table class="wikitable" style="float:right">
					<tbody>
						<tr><th>table table table</th></tr>
					</tbody>
				</table>
			<div>
		`;
		tables();

		expect( document.body.innerHTML ).toMatchSnapshot();
	} );
} );
