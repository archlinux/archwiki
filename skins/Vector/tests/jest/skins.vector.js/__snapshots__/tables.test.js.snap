// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tables doesnt wrap floated tables 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<table class=\\"wikitable\\" style=\\"float:right\\">
					<tbody>
						<tr><th>table table table</th></tr>
					</tbody>
				</table>
			<div>
		</div></div>"
`;

exports[`tables doesnt wrap nested tables 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<div class=\\"noresize\\"><table class=\\"wikitable\\">
					<tbody>
						<tr><th>table table table</th></tr>
						<tr><td><table class=\\"wikitable\\"><tbody><tr><th>table table table</th></tr></tbody></table></td><td></td></tr>
					</tbody>
				</table></div>
			</div>
		"
`;

exports[`tables doesnt wrap tables that already have noresize 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<div class=\\"noresize\\">
					<table class=\\"wikitable\\">
						<tbody>
							<tr><th>table table table</th></tr>
						</tbody>
					</table>
				</div>
			</div>
		"
`;

exports[`tables doesnt wrap tables that are already wrapped 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<div>
					<table class=\\"wikitable\\">
						<tbody>
							<tr><th>table table table</th></tr>
						</tbody>
					</table>
				</div>
			</div>
		"
`;

exports[`tables doesnt wrap tables that are not wikitables 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<table>
					<tbody>
						<tr><th>table table table</th></tr>
						<tr><td><table><tbody><tr><th>table table table</th></tr></tbody></table></td><td></td></tr>
					</tbody>
				</table>
			<div>
		</div></div>"
`;

exports[`tables wraps multiple table with div 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<div class=\\"noresize\\"><table class=\\"wikitable\\">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
				<div class=\\"noresize\\"><table class=\\"wikitable\\">
				</table></div><div class=\\"noresize\\"><table class=\\"wikitable\\">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
				<div class=\\"noresize\\"><table class=\\"wikitable\\">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
			</div>
		"
`;

exports[`tables wraps table with div 1`] = `
"
			<div class=\\"mw-parser-output\\">
				<div class=\\"noresize\\"><table class=\\"wikitable\\">
					<tbody><tr><th>table table table</th></tr></tbody>
				</table></div>
			</div>
		"
`;
